glib/src/gobject/binding_group.rs

```diff
@@ -233,8 +252,159 @@ impl<'a> BindingGroupBuilder<'a> {
     }
 
     // rustdoc-stripper-ignore-next
-    /// Similar to `try_build` but fails instead of panicking.
+    /// Similar to `try_build` but panics instead of failing.
     pub fn build(self) {
         self.try_build().unwrap()
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::prelude::*;
+    use crate::subclass::prelude::*;
+
+    #[test]
+    fn binding_without_source() {
+        let binding_group = crate::BindingGroup::new();
+
+        let source = TestObject::default();
+        let target = TestObject::default();
+
+        assert!(source.find_property("name").is_some());
+        binding_group
+            .bind("name", &target, "name")
+            .flags(crate::BindingFlags::BIDIRECTIONAL)
+            .build();
+
+        binding_group.set_source(Some(&source));
+
+        source.set_name("test_source_name");
+        assert_eq!(source.name(), target.name());
+
+        target.set_name("test_target_name");
+        assert_eq!(source.name(), target.name());
+
+        let err = binding_group.bind("enabled", &target, "name").try_build();
+
+        assert!(err.is_err())
```

Comment:
This fails because the binding group is **not** checking if the types are matching, and it also really can't because the transform functions might make it work for this set of types.

Reply:
@A6GibKm This is still pending

Reply:
Removed the assert.